tools/clang/lib/CodeGen/CGHLSLMSFinishCodeGen.cpp

```diff
@@ -2848,8 +2854,13 @@ void TranslateRayQueryConstructor(HLModule &HLM) {
       llvm::IRBuilder<> Builder(CI);
       llvm::Value *rayFlags =
           Builder.getInt32(SA->GetTemplateArgAnnotation(0).GetIntegral());
-      llvm::Value *Call =
-          Builder.CreateCall(opFunc, {opVal, rayFlags}, pThis->getName());
+      // this could be a nullptr assignment if there is no 2nd template arg
```

Comment:
Is this possible with the current template definition?  Or will the default argument always be here?

Reply:
Comments and code above appear to assume there could be 1 or 2 template arguments, but this code would crash if that's the case, right?  It seems like some adjustment is still necessary for the code and comments to be fully consistent with what's possible at this point.